lib/rules/max-statements.js

```diff
@@ -84,9 +84,18 @@ module.exports = {
          */
         function reportIfTooManyStatements(node, count, max) {
             if (count > max) {
+                const messageEnd = " has too many statements ({{count}}). Maximum allowed is {{max}}.";
+                let name = "This function";
+
+                if (node.id) {
+                    name = `Function '${node.id.name}'`;
+                } else if (node.parent.type === "MethodDefinition" || node.parent.type === "Property") {
+                    name = `Function '${node.parent.key.name}'`;
+                }
```

Comment:
@platinumazure what "this" are you referring to?

@mysticatea I'm not sure what you mean. Properties aren't checked, only functions. Can you explain with an example?